src/pkg/stacks/stacks.go

```diff
@@ -0,0 +1,151 @@
+package stacks
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	cliClient "github.com/DefangLabs/defang/src/pkg/cli/client"
+	"github.com/DefangLabs/defang/src/pkg/modes"
+	"github.com/DefangLabs/defang/src/pkg/term"
+)
+
+type StackParameters struct {
+	Name     string
+	Provider cliClient.ProviderID
+	Region   string
+	Mode     modes.Mode
+}
+
+func Create(params StackParameters) error {
+	if params.Name == "" {
+		return errors.New("stack name cannot be empty")
+	}
+	if !regexp.MustCompile(`^[a-zA-Z0-9]+$`).MatchString(params.Name) {
+		return errors.New("stack name must be alphanumeric")
+	}
+
+	content := Marshal(params)
+	filename := params.Name + ".defangrc"
```

Comment:
This is different than I'd expect. The `.env` convention seems to append the env name to the end: https://github.com/bkeepers/dotenv#customizing-rails